### PR TITLE
[Perf] Add pipeline for Core

### LINKED ISSUE #29716: [Perf] Add pipeline for core
Bootstrap `java - core - perf` pipeline with empty bicep file and `java -jar .\azure-core-perf-1.0.0-beta.1-jar-with-dependencies.jar jsonsend` scenario as a starter.

### PR DESCRIPTION
**Pull request by mikeharder**
- Fixes #29716
- Depends on https://github.com/Azure/azure-sdk-tools/pull/3547
- Validation run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1680623&view=results

### CODE REVIEW

File: eng/pipelines/templates/jobs/perf.yml [hunk 1]
```diff
@@ -37,6 +40,7 @@ resources:
     type: github
     endpoint: Azure
     name: Azure/azure-sdk-tools
+    ref: refs/pull/3547/merge
```
Comment (benbp) [inline eng/pipelines/templates/jobs/perf.yml:43]: Temporary?

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/jobs/perf.yml [hunk 2]
```diff
@@ -23,6 +23,9 @@ parameters:
   - name: additionalArguments
     type: string
     default: ''
+  - name: disableAzureResourceCreation
```
Comment (kasobol-msft) [inline eng/pipelines/templates/jobs/perf.yml:26]: we can keep it if it's useful, but my intention is to eventually deploy resources, so empty template would be as good as this.
For starter I'm planning to add storage account with container accessible via sas url.

**Review by kasobol-msft — APPROVED** [review]

File: eng/pipelines/templates/jobs/perf.yml (hunk 2 above)
Comment (mikeharder) [reply eng/pipelines/templates/jobs/perf.yml]: I will remove this and add the empty bicep file

File: eng/pipelines/templates/jobs/perf.yml (hunk 1 above)
Comment (mikeharder) [reply eng/pipelines/templates/jobs/perf.yml:40]: Yes, this is temporary to validate the changes.  Will be removed when 3547 is merged to main.

**Pull request merged**